File: server_side/src/Logs/LogsCreator.php
<?php

namespace Currency\Logs;

use Currency\CustomInterfaces\ICreator;

class LogsCreator implements ICreator
{
    /**
     * Create instance of ILog
     * @param string $class - class of logs [Error, Warning etc.]
     * @return ILog
     */
    public function create(string $class): ILog
    {
        $namespace = 'Currency\Logs\\';
        $logType = $namespace . $class . 'Log';
        if (class_exists($logType))
            return new $logType();

        return new ErrorLog();
    }
}

File: pc_side/src/logger/logscreator.cpp
#include "logscreator.h"
#include "errorslog.h"
#include "warningslog.h"

LogsCreator::LogsCreator()
{

}

LogsHandler * LogsCreator::create(const std::string logType)
{
   if(logType == "error")
       return new ErrorsLog();
   else if (logType == "warning")
       return new WarningsLog();
   else
       return new ErrorsLog();
    /*
    switch(atoi(logType)) {
     case atoi("errors"):
        return new ErrorsLog();
        break;
    default:
        return new ErrorsLog();
        break;
    }
*/
    //return new (logType)();
   // ErrorsLog *handler = new ErrorsLog();
    //return new ErrorsLog();
    //return new {"ErrorsLog"}();
}
